src/squawk/SquawkAssignment.cpp

```diff
@@ -177,11 +183,26 @@ namespace UKControllerPlugin {
             if (!this->activeCallsigns.UserHasCallsign()) {
                 return false;
             }
-
+            
             return this->GeneralAssignmentNeeded(flightPlan, radarTarget) &&
                 this->storedFlightplans.HasFlightplanForCallsign(flightPlan.GetCallsign()) &&
                 this->storedFlightplans.GetFlightplanForCallsign(flightPlan.GetCallsign())
                     .HasPreviouslyAssignedSquawk();
         }
+
+        /*
+            Checks if the flight plan has basic validations such as destination ICAO being correct.
+        */
+        bool SquawkAssignment::NeedsSquawkFlightPlan(
+            const EuroScopeCFlightPlanInterface & flightPlan,
+            const EuroScopeCRadarTargetInterface & radarTarget
+        ) const
+        {
+            if (flightPlan.GetOrigin().empty() || flightPlan.GetOrigin() == " " || flightPlan.GetOrigin() == "NOFP" || flightPlan.GetOrigin().length() >= 5) {

```

Comment:
Could possibly simplify this by trimming the string of whitespace on either side and then just doing `.length !=4` - that would catch the first two and last cases :)